### PR TITLE
chore: slice refactor

### PR DESCRIPTION
**Pull request by tuncaulubilge**
"Small" refactor on tiles and slices, changed couple of things and tidied showcases and tests

- Mocked tiles inside slice tests, as we already have tile specific tests. This cleaned up lots of snaps
- Update the api for each slice, so they just take a prop called "slice" instead of things like "lead" "support1" etc
- Instead of exporting each slice separately, now they are all exported as a map. This way, section page does not need to know about each slice we support, it just consumes the map
- Minor refactors in tests to make our fixtures consistent

### CODE REVIEW

File: packages/edition-slices/__tests__/shared-slices.base.js [hunk 1]
```diff
@@ -22,26 +26,38 @@ import {
   SecondaryTwoNoPicAndTwoSlice
 } from "../src/slices";
 
+jest.mock("../src/tiles", () => ({
+  TileA: "TileA",
```
Comment (aronshamash) [inline packages/edition-slices/__tests__/shared-slices.base.js]: can we generate programmatically using `fs` ?

File: packages/edition-slices/edition-tiles.showcase.js [hunk 2]
```diff
@@ -31,145 +27,112 @@ const preventDefaultedAction = decorateAction =>
 export default {
```
Comment (aronshamash) [inline packages/edition-slices/edition-tiles.showcase.js]: We should map over an object literal to generate this maybe ?

File: packages/fixture-generator/src/index.ts [hunk 3]
```diff
@@ -16,8 +16,9 @@ import mockEditionSlice, {
   mockLeadersSlice,
   mockSecondaryOneSlice,
   mockSecondaryFourSlice,
+  mockSecondaryTwoAndTwoSlice,
   mockSecondaryTwoNoPicAndTwoSlice,
-  mockList2AndSixNoPicSlice
+  mockListTwoAndSixNoPicSlice
```
Comment (aronshamash) [inline packages/fixture-generator/src/index.ts]: could be a good time to alphabetise here ?

File: packages/pages/src/section/section.js [hunk 4]
```diff
@@ -1,79 +1,15 @@
...
-  ListTwoAndSixNoPicSlice,
-  SecondaryOneSlice,
-  SecondaryFourSlice
-} from "@times-components/edition-slices";
+import sliceMap from "@times-components/edition-slices";
 import SectionItemSeparator from "./section-item-separator";
 import styles from "./styles";
 
 const renderSlice = (slice, onPress) => {
-  switch (slice.name) {
```
Comment (aronshamash) [inline packages/pages/src/section/section.js]: nice :)

Comment (tuncaulubilge) [reply packages/pages/src/section/section.js]: :) this was pretty much the goal of this whole pr

File: packages/edition-slices/__tests__/shared-slices.base.js (hunk 1 above)
Comment (tuncaulubilge) [reply packages/edition-slices/__tests__/shared-slices.base.js]: Done :+1:

**Pull request merged**